d2renderers/d2svg/d2svg.go

```diff
@@ -51,34 +53,24 @@ var TooltipIcon string
 var LinkIcon string
 
 //go:embed style.css
-var styleCSS string
-
-//go:embed sketchstyle.css
-var sketchStyleCSS string
+var baseStylesheet string
 
 //go:embed github-markdown.css
 var mdCSS string
 
 type RenderOpts struct {
-	Pad    int
-	Sketch bool
+	Pad         int
+	Sketch      bool
+	ThemeID     int64
+	DarkThemeID int64
 }
 
-func setViewbox(writer io.Writer, diagram *d2target.Diagram, pad int) (width int, height int) {
+func dimensions(writer io.Writer, diagram *d2target.Diagram, pad int) (width, height int, topLeft, bottomRight d2target.Point) {
 	tl, br := diagram.BoundingBox()
 	w := br.X - tl.X + pad*2
 	h := br.Y - tl.Y + pad*2
-	// TODO minify
 
-	// TODO background stuff. e.g. dotted, grid, colors
-	fmt.Fprintf(writer, `<?xml version="1.0" encoding="utf-8"?>
-<svg
-id="d2-svg"
-style="background: white;"
-xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
-width="%d" height="%d" viewBox="%d %d %d %d">`, w, h, tl.X-pad, tl.Y-pad, w, h)
-
-	return w, h
+	return w, h, tl, br
```

Comment:
tl and br should subtract pad here. it's confusing to give w/h with pad and tl and br without pad